CHANGELOG

```diff
@@ -1,7 +1,18 @@
 devel
 -----
 
-* fix compation behaviour of followers
+* added option `--all-databases` to arangodump
+
+  When set to true, this makes arangodump dump all available databases
+  the current user has access to. The option `--all-databases` cannot be
+  used in combination with the option `--server.database`. 
```

Comment:
What about `--collection`?

For a complete dump, you don't need it anyway. Name-based restrictions but across multiple databases would be weird, and I also don't see an obvious solution how one could specify which collections to dump from which database on the command line... Perhaps the masking feature can be extended to work for multiple databases, so that you can `exclude` certain collections via a config file?

Also see my comment here: https://github.com/arangodb/planning/issues/3681#issuecomment-467337095

Reply:
I think `--collection` can be used together with `--all-databases true`.
It may make sense if the use case is multi-tenancy and you have an `orders` collection in each database... Why not dump just these `orders` collections across all databases?

Reply:
Not sure about maskings... Will the feature need some extensions to work with `--all-databases`, or can we implement that completely independently?

Reply:
Don't know, but I assume the same masking configuration would be applied to each database? Then it's basically the same as with `--collection`: it makes sense for multi-tenancy where each database has the same collection structure.

If you want to dump different collections per database, then we would need to extend the maskings feature somehow, but that could be done separately as far as I can tell.